ccloader/js/mod.js

```diff
@@ -149,6 +183,79 @@ export class Mod {
 		this.runtimeAssets[original] = newPath;
 	}
 
+	/**
+	 * Returns the full path to a resource
+	 * @param {string} relativePath to resource with mod folder as base
+	 * @returns {string} 
+	 */
+	absolutePath(relativePath) {
+		const basePath = this._getBaseName(this.file); 
+		if (!relativePath) {
+			return this._normalizePath(basePath);
+		}
+
+		if (navigator.platform === 'Win32') {
+			// convert all \ to / and remove all duplicate //
+			relativePath = relativePath.replace(/\\/g, path.sep).replace(/\/\//g, path.sep);
+		}
+
+		// should protect against escaping mod directory
+		relativePath = path.normalize('/' + relativePath);
+		return this._normalizePath(basePath + relativePath);
+	}
+
+
+	/**
+	 * 
+	 * @param {string} relativePath to resource with mod folder as base
+	 * @returns {boolean} 
+	 */
+	hasResource(relativePath) {
+		if (!this.ready) {
+			return false;
+		}
+
+		for (const asset of this.assets) {
+			if (asset.endsWith(relativePath)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * 
+	 * @param {string} relativePath to resource with mod folder as base
+	 * @returns {any} 
+	 */
+	async getResource(relativePath) {
+		if (!this.ready) {
+			throw Error(`Cannot get resource from Mod "${this.manifest.name}". Mod is not ready.`);
+		}
+
+		// prevents attempts to get resources they don't have
+		if (!this.hasResource(relativePath)) {
+			const fullUrl = this.path + path.normalize('/' + relativePath);
```

Comment:
If `absolutePath` is changed to include `location.origin` then that function should be preferred and `this.path` should be removed.

Reply:
absolutePath is not for internal use only.